x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/WordBoundaryChunkingSettings.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.chunking;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.inference.ChunkingSettings;
+import org.elasticsearch.inference.ChunkingStrategy;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.ServiceUtils;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+public class WordBoundaryChunkingSettings implements ChunkingSettings {
+    public static final String NAME = "WordBoundaryChunkingSettings";
+    private static final ChunkingStrategy STRATEGY = ChunkingStrategy.WORD;
+    protected final int maxChunkSize;
+    protected final int overlap;
+
+    public WordBoundaryChunkingSettings(Integer maxChunkSize, Integer overlap) {
+        this.maxChunkSize = maxChunkSize;
+        this.overlap = overlap;
+    }
+
+    public WordBoundaryChunkingSettings(StreamInput in) throws IOException {
+        maxChunkSize = in.readInt();
+        overlap = in.readInt();
+    }
+
+    public static WordBoundaryChunkingSettings fromMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+        Integer maxChunkSize = ServiceUtils.extractRequiredPositiveInteger(
+            map,
+            ChunkingSettingsOptions.MAX_CHUNK_SIZE.toString(),
+            ModelConfigurations.CHUNKING_SETTINGS,
+            validationException
+        );
+
+        Integer overlap = null;
+        if (maxChunkSize != null) {
+            overlap = ServiceUtils.extractRequiredPositiveIntegerLessThanOrEqualToMax(
+                map,
+                ChunkingSettingsOptions.OVERLAP.toString(),
+                maxChunkSize / 2,
+                ModelConfigurations.CHUNKING_SETTINGS,
+                validationException
+            );
+        }
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+
+        return new WordBoundaryChunkingSettings(maxChunkSize, overlap);
```

Comment:
Same comment about leniency when paring user supplied config, check for unknown options.